extensions/hibernate-validator/deployment/src/main/java/io/quarkus/hibernate/validator/deployment/HibernateValidatorProcessor.java

```diff
@@ -177,6 +181,8 @@ void registerAdditionalBeans(HibernateValidatorRecorder hibernateValidatorRecord
                 .requiresContainerServices()
                 .defaultScope(BuiltinScope.DEPENDENT).build());
 
+        additionalBeans.produce(new AdditionalBeanBuildItem(ConstraintMappingProvider.class));
```

Comment:
This doesn't do anything I think because it's an interface. You would need to point it to an actual implementation but that won't work in practice because implementations are user-provided.

If you wanted to do that, you would have to use the Jandex index to get to the implementations of this interface and then register them as beans (and probably define a default scope).

But I don't think you need that, more on that in my general review.